lib/queues.js

```diff
@@ -260,13 +270,62 @@ class Queue {
   runOnce() {
     assert(!this._processing, 'Cannot call once while continuously processing jobs');
     const hasHandlers = (this._handlers.size > 0);
+
     if (hasHandlers) {
 
       debug('Waiting for jobs on queue %s', this.name);
-      return this._getClientPromise.then(client => this._reserveAndProcess(client));
+
+      if (process.env.NODE_ENV === 'test' && this._delayedJobs.size > 0) {
```

Comment:
What about keeping the if statement for `delayedJobs.size > 0` to avoid going through this logic when there are no delayed jobs?